### ISSUE TITLE
Minishift openshift version list should show supported list

### ISSUE THREAD

**Issue by praveenkumar** [minishift/minishift#1244]
Currently Minishift version list only show version >= our default set version (as below) but what we want to show openshift versions which can be provisioned (at least v1.4.0 onwards).

```
$ ./minishift openshift version list
The following OpenShift versions are available: 
	- v3.6.0
	- v3.6.0-alpha.0
	- v3.6.0-alpha.1
	- v3.6.0-alpha.2
	- v3.6.0-rc.0
	- v3.7.0-alpha.0
```

**Comment by LalatenduMohanty** [comment]
This is a code issue. We only support versions greater than what we mention in Makefile https://github.com/minishift/minishift/blob/master/pkg/minishift/openshift/version/openshift_versions.go#L77

**Comment by LalatenduMohanty** [comment]
I suggest we introduce a constant for minimum supported version and then we can refer it for this.

**Comment by LalatenduMohanty** [comment]
Also we need a integration test for this.

**Comment by hferentschik** [comment]
I am pretty sure this a regression. At some point of time we did the right thing. @praveenkumar was implementing the code for this.

@agajdosi another one released to OpenShift versions. :-)

**Comment by praveenkumar** [comment]
> I am pretty sure this a regression. At some point of time we did the right thing. @praveenkumar was implementing the code for this.

Yes it's just part of regression, we did fix it some time back :(

**Comment by hferentschik** [comment]
> Yes it's just part of regression, we did fix it some time back :(

This should be testable via unit tests as well, right? Tests tests tests ! ;-)

That said, it is also cheap to test from an integration test pov, since one does not even need a running VM. One way or the other, a pull request should take care that this regression won't occur again.

**Comment by hferentschik** [comment]
Merged via pull request #1255
